src/include/basil.h

```diff
@@ -69,193 +68,193 @@
-#define BASIL_ELM_RESPONSEDATA	"ResponseData"	/* BASIL_ELM_RESPONSE */
-#define BASIL_ELM_RESERVED	"Reserved"	/* BASIL_ELM_RESPONSEDATA */
-#define BASIL_ELM_CONFIRMED	"Confirmed"	/* BASIL_ELM_RESPONSEDATA */
-#define BASIL_ELM_RELEASED	"Released"	/* BASIL_ELM_RESPONSEDATA */
-#define BASIL_ELM_ENGINE	"Engine"	/* BASIL_ELM_RESPONSEDATA */
-#define BASIL_ELM_INVENTORY	"Inventory"	/* BASIL_ELM_RESPONSEDATA */
-#define BASIL_ELM_NETWORK	"Network"	/* BASIL_ELM_RESPONSEDATA */
-#define BASIL_ELM_NODEARRAY	"NodeArray"	/* BASIL_ELM_INVENTORY */
-#define BASIL_ELM_NODE		"Node"		/* BASIL_ELM_NODEARRAY */
-#define BASIL_ELM_ACCELPARAMARRAY "AccelParamArray" /* BASIL_ELM_RESERVEPARAM */
-#define BASIL_ELM_ACCELPARAM	"AccelParam"	/* BASIL_ELM_ACCELPARAMARRAY */
-#define BASIL_ELM_ACCELERATORARRAY "AcceleratorArray" /* BASIL_ELM_NODE */
-#define BASIL_ELM_ACCELERATOR   "Accelerator"	/* BASIL_ELM_ACCELERATORARRAY */
-#define BASIL_ELM_ACCELERATORALLOC "AcceleratorAllocation"
-/* BASIL_ELM_ACCELERATOR */
-#define BASIL_ELM_SEGMENTARRAY	"SegmentArray"	/* BASIL_ELM_NODE */
-#define BASIL_ELM_SEGMENT	"Segment"	/* BASIL_ELM_SEGMENTARRAY */
-#define BASIL_ELM_PROCESSORARRAY "ProcessorArray"
-/* BASIL_ELM_SEGMENT */
-#define BASIL_ELM_PROCESSOR	"Processor"	/* BASIL_ELM_PROCESSORARRAY */
-#define BASIL_ELM_PROCESSORALLOC "ProcessorAllocation"
-/* BASIL_ELM_PROCESSOR */
-#define BASIL_ELM_MEMORYARRAY	"MemoryArray"	/* BASIL_ELM_SEGMENT */
-#define BASIL_ELM_MEMORY	"Memory"	/* BASIL_ELM_MEMORYARRAY */
-#define BASIL_ELM_MEMORYALLOC	"MemoryAllocation" /* BASIL_ELM_MEMORY */
-#define BASIL_ELM_LABELARRAY	"LabelArray"	/* BASIL_ELM_SEGMENT */
-#define BASIL_ELM_LABEL		"Label"		/* BASIL_ELM_LABELARRAY */
-#define BASIL_ELM_RSVNARRAY	"ReservationArray" /* BASIL_ELM_INVENTORY */
-/* BASIL_ELM_RESPONSEDATA */
-#define BASIL_ELM_RESERVATION	"Reservation"	/* BASIL_ELM_RSVNARRAY */
-#define BASIL_ELM_APPARRAY	"ApplicationArray"/* BASIL_ELM_RESERVATION */
-/* BASIL_ELM_RESPONSEDATA */
-#define BASIL_ELM_APPLICATION	"Application"	/* BASIL_ELM_APPARRAY */
-#define BASIL_ELM_CMDARRAY	"CommandArray"	/* BASIL_ELM_APPLICATION */
-#define BASIL_ELM_COMMAND	"Command"	/* BASIL_ELM_CMDARRAY */
-#define BASIL_ELM_RSVD_NODEARRAY	"ReservedNodeArray"
-/* BASIL_ELM_RESERVED */
-#define BASIL_ELM_RSVD_NODE		"ReservedNode"
-/* BASIL_ELM_RSVD_NODEARRAY */
-#define BASIL_ELM_RSVD_SGMTARRAY	"ReservedSegmentArray"
-/* BASIL_ELM_RSVD_NODE */
-#define BASIL_ELM_RSVD_SGMT		"ReservedSegment"
-/* BASIL_ELM_RSVD_SGMTARRAY */
-#define BASIL_ELM_RSVD_PROCARRAY	"ReservedProcessorArray"
-/* BASIL_ELM_RSVD_SGMT */
-#define BASIL_ELM_RSVD_PROCESSOR	"ReservedProcessor"
-/* BASIL_ELM_RSVD_PROCARRAY */
-#define BASIL_ELM_RSVD_MEMARRAY		"ReservedMemoryArray"
-/* BASIL_ELM_RSVD_SGMT */
-#define BASIL_ELM_RSVD_MEMORY		"ReservedMemory"
-/* BASIL_ELM_RSVD_MEMARRAY */
-
+#define BASIL_ELM_MESSAGE	"Message"		/* All elements */
+#define BASIL_ELM_REQUEST	"BasilRequest"		/* Top level */
```

Comment:
Hi Arun,
   Comments in these preprocessor, tells about the parent element name to which they belong in xml response of apbasil.

Regards